### PR TITLE
Periods crash in muon analysis

### LINKED ISSUE #19975: Periods in muon analysis causing a crash.
The following instructions caused the error 'Fit Algorithm failed. Unknown property search object InputWorkspace_1' 

1. Change facility to ISIS, default instrument to MUSR
2. Open the Muon Analysis interface
3. Click Manage Directiories and add Mantid training course data to Data Search Directories
4. Browse... and open MUSR00015189.nxs from the training course data
5. Switch between the three options in Group/Group Pair combo box.
6. Go to Settings tab, make sure multiple fitting is disabled and TF Asymmetry enabled
7. Go to Data Analysis tab
8. There is thee options in Data/Workspace combo box. Switch between them and the number of options increases...
9. When you have all six options in the Data/Workspace combo box, add GausOsc fitting function within the CompositeFunction
10. Fit. An error dialog with the aforementioned message appears

There are two bugs here. The first is that the periods are not loaded into data analysis correctly in the first place. The second is that the fit is trying to fit to multiple periods (the bugs do not occur if there is only a single period).

### PR DESCRIPTION
**Pull request by AnthonyLim23**
Muon analysis was crashing when multiple period data was loaded into the single fit. 

**To test:**

The following instructions caused the error 'Fit Algorithm failed. Unknown property search object InputWorkspace_1'

1. Open the Muon Analysis interface
2. Load MUSR00015189.nxs 
3. Switch between the options in Group/Group Pair combo box on the home tab (two is enough).
4. Go to Settings tab, make sure multiple fitting is disabled
5. Go to Data Analysis tab
6. Add a function (e.g. flatBackground) and do a fit. It should work.
7. Change the workspace and do another fit. These should also (now) work. 

Ignore the message "cannot reach the specified tolerance in F", this just means that the fit was poor. 

Fixes #19975. 

**Release Notes** 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: MantidQt/MantidWidgets/src/MuonFitPropertyBrowser.cpp [hunk 1]
```diff
@@ -380,6 +380,27 @@ void MuonFitPropertyBrowser::enumChanged(QtProperty *prop) {
...
+      }
+      end = option.find_first_of(";");
+      QString selectedPeriod = QString::fromStdString(option.substr(0, end));
+      // turn on only the relevant box
+      for (auto iter = m_periodBoxes.constBegin();
+           iter != m_periodBoxes.constEnd(); ++iter) {
+        if (selectedPeriod == iter.key()) {
+          m_boolManager->setValue(iter.value(), true);
+        } else {
+          m_boolManager->setValue(iter.value(), false);
```
Comment (PranavBahuguna) [inline MantidQt/MantidWidgets/src/MuonFitPropertyBrowser.cpp]: Why not just do something like this?

```
someBool = selectedPeriod == iter.key()
m_boolManager->setValue(iter.value(), true);
```

Or even just:

```
m_boolManager->setValue(iter.value(), selectedPeriod == iter.key());
```

**Review by PranavBahuguna — CHANGES_REQUESTED** [review]
Changes work as expected and the bug no longer appears when following the testing instructions. Code also looks good, just address my one comment and its probably good to go.

File: MantidQt/MantidWidgets/src/MuonFitPropertyBrowser.cpp (hunk 1 above)
Comment (AnthonyLim23) [inline MantidQt/MantidWidgets/src/MuonFitPropertyBrowser.cpp]: do you mean

```
bool state = (selectedPeriod == iter.key()) ? true : false;
m_boolManager->setValue(iter.value(),,state);
```

Comment (PranavBahuguna) [inline MantidQt/MantidWidgets/src/MuonFitPropertyBrowser.cpp]: I don't think the conditional is necessary, it should compile just fine without it.

Comment (AnthonyLim23) [inline MantidQt/MantidWidgets/src/MuonFitPropertyBrowser.cpp]: I will try it after lunch.

**Review by PranavBahuguna — APPROVED** [review]
All looks good, :shipit: once checks pass

**Pull request merged**